internal/descriptor/registry.go

```diff
@@ -115,13 +115,21 @@ type Registry struct {
 
 	// recursiveDepth sets the maximum depth of a field parameter
 	recursiveDepth int
+
+	// annotationMap is used to check for duplicate HTTP annotations
+	annotationMap map[annotationIdentifier]bool
```

Comment:
Nit, but this could be slightly more efficient with an empty struct value:
```suggestion
	annotationMap map[annotationIdentifier]struct{}
```